WalletWasabi/WalletWasabi.csproj

```diff
@@ -81,4 +81,13 @@
     <RuntimeHostConfigurationOption Include="System.Globalization.Invariant" Value="true" />
   </ItemGroup>
 
+  <Target Name="EmbedLocal" BeforeTargets="PrepareForBuild">
+    <Exec ContinueOnError="True" Command="git log --pretty=format:'%H' -n 1 > &quot;$(ProjectDir)/GitInfo/git-info-commit.txt&quot;" />
+    <Exec ContinueOnError="True" Command="git rev-parse --abbrev-ref HEAD > &quot;$(ProjectDir)/GitInfo/git-info-branch.txt&quot;" />
```

Comment:
I think this PR is great, but I don't know if it's important enough to introduce things like this that we aren't using :/ I kinda feel like it can only cause issues in the future. Anyhow, these are my thoughts, it's up to you if you want to do this.

Reply:
@nopara73 yes, but the alternative is us importing another nuget dependency. This one is a msbuild command supported since VS 2017  and honestly we should use more of msbuild to automate some stuff more imo :)

Reply:
Also the git commands are compatible in most of the git versions we use so i think it should work everywhere

Reply:
I agree here. This shouldn't be part of the project and it is looking for troubles. What happens if I don't have `git` in my environment? How does this behave?

Reply:
If GitVersioning doesn't use any magic, then I'd vote for copypasting the relevant parts into Wasabi's codebase: https://github.com/dotnet/Nerdbank.GitVersioning/

See https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Nito, https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Mono, https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Gma/QrCodeNet/Encoding as examples.

Reply:
> If GitVersioning doesn't use any magic, then I'd vote for copypasting the relevant parts into Wasabi's codebase: https://github.com/dotnet/Nerdbank.GitVersioning/
> 
> See https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Nito, https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Mono, https://github.com/zkSNACKs/WalletWasabi/tree/master/WalletWasabi/Gma/QrCodeNet/Encoding as examples.

It has a huge msbuild hackery so i rather not. Trust me this is the cleanest way of doing things...

> I agree here. This shouldn't be part of the project and it is looking for troubles. What happens if I don't have `git` in my environment? How does this behave?

It just continues the build with a warning, nothing more, nothing less :) but there wont be any hash displayed in About view..